### PR TITLE
cleanup & updates

### PR DESCRIPTION
**Pull request by notslang**
- Switched to argparse, which handles most parameter validation and `--help` page generation (allowing us to get rid of a bunch of code)
- Reformatted the code/docs via tidy-markdown, standard-format, and fixpack
- Used csv-write-stream to get rid of `makeCSVStream()`... Alternatively we could just get rid of the `--format` arg altogether and tell people to do `lastfmexport -u USER | csv-write > user.csv`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,32 +1,33 @@
-lastfmexport
-============
```
Comment (maxkueng) [inline README.md:2]: What's wrong with this?

**Comment by maxkueng** [comment]
@slang800 Thanks for all the cool commits! I have a few questions though.
- `rucola` provides a few cool features that `argparser` does not seem to
  have. It allows you to create a config file with all the same options that
  the at are available as CLI arguments, in any common location where you'd
  typically put config files (e.g.  `~/.lastfmexportrc`). Additionally it also
  allows you to use environment variables like
  `LASTFMEXPORT_USERNAME=somagroovesalad lastfmexport` (or put it in
  `.bashrc`).
  
  `argparser` has a nice declarative way to define the arguments and
  validation and usage text generation is cool and all but I don't think that
  the benefits outweigh recreating those features (config files and
  environment variables) within the app.
- `csv-write-stream` is more code than the current `makeCSVStream()` and it
  doesn't seem to produce a better file or increase performance. In fact, we
  currently produce a tab-separated file and `csv-write-stream` produces a
  comma-separated file. So that would be a breaking change. We could use the
  `separator` option in `csv-write-stream` but I'm not sure whether it's
  really necessary to use the module. 
- I'm okay with changing the code to use tabs instead of spaces and using
  tools to check the code quality would be awesome. However, I'm absolutely
  against using the "standard" js format. Apart from that I think calling the
  thing "standard" is a total asshole-move, I want my code to have semicolons.

File: README.md (hunk 1 above)
Comment (notslang) [inline README.md:2]: Nothing's inherently wrong with it, besides perhaps the need to change the number of `=` chars whenever the title changes, resulting in a larger diff and the difference in syntax from deeper headers (h3-h6). It's just what [tidy-markdown](https://github.com/slang800/tidy-markdown) does automatically when it's cleaning up markdown .

**Comment by notslang** [comment]
> `rucola` ... allows you to create a config file with all the same options that the at are available as CLI arguments, in any common location where you'd typically put config files (e.g. `~/.lastfmexportrc`). Additionally it ... allows you to use environment variables

I'd consider config files to be an antipattern, especially for a tool this small. Not only does it add more file system calls (even when a config file isn't in use), but if the user is unaware of the existence of a config file (like if one was created months ago and forgotten about) then it can result in unexpected behavior. And finally, having a config file introduces a whole new class of problems that come from managing changes in the schema of the config file between versions. If you're running lastfmexport so often that typing in args becomes a significant waste of time, then you should put the full command in bash script (just paste it into `lastfmscrape.sh`) or create an [alias](http://www.tldp.org/LDP/abs/html/aliases.html).

> csv-write-stream is more code than the current `makeCSVStream()` and it doesn't seem to produce a better file or increase performance.

The idea is to remove duplicated code so this project doesn't need to maintain its own CSV writing function. The module may contain more lines of code, but it's depended upon by many authors and has gotten more review than `makeCSVStream()`. Also, it should be a bit faster because of the way it precompiles a writer function. I suspect that the network requests are the primary bottleneck in this module, so testing it without isolating it wouldn't show much, but on average it should be an improvement.

> We currently produce a tab-separated file and csv-write-stream produces a comma-separated file. So that would be a breaking change. We could use the separator option in csv-write-stream but I'm not sure whether it's really necessary to use the module.

CSV is "comma separated values", so I think it would make sense to switch. Alternatively, getting rid of the `--format` option entirely and only outputting ndjson would be a good idea. Then we could just give instructions on how to convert it into whatever format people want by piping to [jq](https://stedolan.github.io/jq/) or [csv-write-stream](https://github.com/maxogden/csv-write-stream).

> I'm absolutely against using the "standard" js format. Apart from that I think calling the thing "standard" is a total asshole-move, I want my code to have semicolons.

Haha, yeah - the name is pretty awful, though not having semicolons is actually quite nice when you get used to it. If you want, you can merge all but the last 2 commits, and then I'll rebase the csv-write-stream part without the standard.js conversion.

**Pull request closed without merging**